src/urllib3/connection.py

```diff
@@ -612,63 +613,111 @@ def set_cert(
         self.ca_cert_data = ca_cert_data
 
     def connect(self) -> None:
-        sock: socket.socket | ssl.SSLSocket
-        self.sock = sock = self._new_conn()
-        server_hostname: str = self.host
-        tls_in_tls = False
-
-        # Do we need to establish a tunnel?
-        if self._tunnel_host is not None:
-            # We're tunneling to an HTTPS origin so need to do TLS-in-TLS.
-            if self._tunnel_scheme == "https":
-                # _connect_tls_proxy will verify and assign proxy_is_verified
-                self.sock = sock = self._connect_tls_proxy(self.host, sock)
-                tls_in_tls = True
-            elif self._tunnel_scheme == "http":
-                self.proxy_is_verified = False
-
-            # If we're tunneling it means we're connected to our proxy.
-            self._has_connected_to_proxy = True
-
-            self._tunnel()  # type: ignore[attr-defined]
-            # Override the host with the one we're requesting data from.
-            server_hostname = self._tunnel_host
-
-        if self.server_hostname is not None:
-            server_hostname = self.server_hostname
-
-        is_time_off = datetime.date.today() < RECENT_DATE
-        if is_time_off:
-            warnings.warn(
-                (
-                    f"System time is way off (before {RECENT_DATE}). This will probably "
-                    "lead to SSL verification errors"
-                ),
-                SystemTimeWarning,
+        # Today we don't need to be doing this step before the /actual/ socket
+        # connection, however in the future we'll need to decide whether to
+        # create a new socket or re-use an existing "shared" socket as a part
+        # of the HTTP/2 handshake dance.
+        probe_http2_port: int | None
+        if self._tunnel_host is None:
+            probe_http2_host = self.host
+            probe_http2_port = self.port
+        else:
+            probe_http2_host = self._tunnel_host
+            probe_http2_port = self._tunnel_port
+        if probe_http2_port is None:
+            probe_http2_port = 443
```

Comment:
This is clearly a nit, but I find making the first branch of the `if` related to the condition clearer. And I like to separate different conditions by a newline. Feel to ignore me!

```suggestion
        probe_http2_port: int | None
        if self._tunnel_host is not None:
            probe_http2_host = self._tunnel_host
            probe_http2_port = self._tunnel_port
        else:
            probe_http2_host = self.host
            probe_http2_port = self.port

        if probe_http2_port is None:
            probe_http2_port = 443
```